tests/acceptance/stepDefinitions/sharingContext.js

```diff
@@ -160,14 +162,66 @@ const checkIsListedInAutoComplete = function (type, displayName) {
         displayNameWithType = displayName + '\nGroup'
       }
       return itemsList.includes(displayNameWithType)
-    })
+    }
+    )
+}
+
+const getUsersMatchingPattern = function (pattern) {
+  // check if all created users that contain the pattern either in the display name or the username
+  // are listed in the autocomplete list
+  // in both cases the display name should be listed
+  const users = userSettings.getCreatedUsers()
+  const usersID = Object.keys(users)
+  return usersID.filter(
+    id => users[id].displayname.toLowerCase().includes(pattern) || id.includes(pattern)
+  ).map(
+    id => users[id].displayname
+  )
+}
+
+const getGroupsMatchingPattern = function (pattern) {
+  const groups = userSettings.getCreatedGroups()
+  const groupMatchingPattern = groups.filter(group => group.toLowerCase().includes(pattern))
+  return groupMatchingPattern
+}
+
+const getUsersWithThePattern = async function (pattern, alreadySharedUserOrGroup, userOrGroup) {
+  const itemsList = await client.page.FilesPageElement.sharingDialog().getShareAutocompleteItemsList()
+  const currentUserDisplayName = userSettings.getDisplayNameForUser(client.globals.currentUser)
+  const usersMatchingPattern = getUsersMatchingPattern(pattern).filter(
+    displayName => {
+      return displayName !== currentUserDisplayName && displayName !== alreadySharedUserOrGroup
+    }
+  )
+  const groupPostfix = client.page.FilesPageElement.sharingDialog().getGroupSharePostfix()
+  const userPostfix = client.page.FilesPageElement.sharingDialog().getUserSharePostfix()
+  const usersNotFound = _.difference(
+    usersMatchingPattern.map(name => name + userPostfix),
+    itemsList
+  )
+  assert.strictEqual(usersNotFound.length, 0, `could not find ${usersNotFound} in the itemsList`)
```

Comment:
Please split these into two functions:
1. that asserts the given users are listed in the auto-completion list.
2. that asserts the given groups are listed in the auto-completion list.

And, also adjust the other function (the one *without* `exclude`) and this function accordingly.